### DISCUSSION TITLE
Custom.xaml如何在域名获取

### DISCUSSION THREAD

**Discussion by Ad-closeNN** [Meloong-Git/PCL#4298] [提问] [answered]
弄到了个域名，但是不知道咋弄这个: `在PCL自定义主页链接输入我的域名后能直接获取到同目录下的Custom.xaml`
像别人投稿主页的那种直接输入xxx.com而不是xxx.com/Custom.xaml，不知道咋搞（（
~~HTML纯小白qwq~~

**Comment by hejiehao** [comment]
重定向
我不是很确定 PCL 支不支持重定向

**Comment by Silverteal** [comment, marked as answer]
服务器的配置中应该可以选择“默认页面”。
如果使用托管的服务，且服务商没有提供此设置，可以尝试将文件名改为index.html（嗯...离谱，但未必不行）

**Reply by Silverteal** [reply]
上面这位提到的301跳转应该也行，但html跳转肯定是不行的

**Comment by wuliaodexiaoluo** [comment]
写 301 重定向，PCL 是支持处理重定向请求的
![QQ_1721362883226](https://github.com/user-attachments/assets/b6287ab7-96c7-4366-a6ee-66b8e743151b)
![image](https://github.com/user-attachments/assets/67516eaf-5929-43b6-84b7-f97c7c6c1f6f)
![image](https://github.com/user-attachments/assets/8c80070b-be23-4fd4-8b44-2659b4e2cb20)

**Reply by wuliaodexiaoluo** [reply]
>如果使用托管的服务，且服务商没有提供此设置，可以尝试将文件名改为index.html（嗯...离谱，但未必不行）

根据测试（上图）没有填写的情况下是默认请求 version 这个文件，所以应该改成 version

**Reply by Silverteal** [reply]
据我所知 version 是 PCL 内置的主页版本协议，并不用于提供内容。而大部分服务器的默认配置是：当访问一个目录时，尝试返回其 index.html，和 PCL 没关系

**Comment by Ad-closeNN** [comment]
301 重定向听着不知道可以不可以（~~用的 cloudns 域名 cloudflare 操作，没试过 301~~）
**不过将 Custom.xaml 命名为 index.html 确实可行**

**Reply by Silverteal** [reply]
Cloudflare可以直接用它的批量重定向，或者页面规则，总之带有“URL转发”“重定向”“默认页面”之类的选项应该都可以实现这个效果

**Comment by MoYuan-CN** [comment]
参考你的服务器面板或者 Nginx 配置
你可以配置无访问路由时读取 Custom.xaml 并返回，配置访问 `/version` 时读取你的版本文件并返回
不太建议使用上述改名方案，因为这样操作并不标准，虽然 PCL 不获取文件名

**Comment by Light-Beacon** [comment]
301 302 重定向都可以，你index.html写这玩意也可以，总之你只需要让他http返回主页的xaml文件内容就行，~~你甚至可以架一个后端上去~~
(过来人的经验)
